Sources/XcodeGenKit/ProjectGenerator.swift

```diff
@@ -41,9 +41,10 @@ public class ProjectGenerator {
 
         func getBuildEntry(_ buildTarget: Scheme.BuildTarget) -> XCScheme.BuildAction.Entry {
 
-            let targetReference = pbxProject.objects.nativeTargets.referenceValues.first { $0.name == buildTarget.target }!
+            let predicate: (PBXTarget) -> Bool = { $0.name == buildTarget.target }
+            let targetReference = pbxProject.objects.nativeTargets.referenceValues.first{ predicate($0 as PBXTarget) } ?? pbxProject.objects.legacyTargets.referenceValues.first{ predicate($0 as PBXTarget) }!
```

Comment:
the next version of xcproj will have a helper function for returning a target by name